src/Stratis.Bitcoin/Connection/ProvenHeadersConnectionManagerBehavior.cs

```diff
@@ -31,16 +31,24 @@ public ProvenHeadersConnectionManagerBehavior(IConnectionManager connectionManag
         /// <inheritdoc />
         protected override async Task OnHandshakedAsync(INetworkPeer peer)
         {
-            // Require from height is the highest between activation height and last checkpoint height.
-            int lastCheckpointHeight = this.checkpoints.GetLastCheckpointHeight();
-            int activationHeight = (this.network.Consensus.Options as PosConsensusOptions).ProvenHeadersActivationHeight;
+            if (this.CanPeerProcessProvenHeaders(peer))
+            {
+                // Require from height is the highest between activation height and last checkpoint height.
+                int lastCheckpointHeight = this.checkpoints.GetLastCheckpointHeight();
+                int activationHeight = (this.network.Consensus.Options as PosConsensusOptions).ProvenHeadersActivationHeight;
```

Comment:
I don't think `this.network.Consensus.Options` can ever be `null`. Use direct cast, not `as`, i.e. `(PosConsensusOptions )this.network.Consensus.Options`

Reply:
This will indeed go away soon but for now its correct yes, we should not send PH to nodes under that protocol version